### PR TITLE
Check device ID when initializing a thing handler

### LINKED ISSUE #34: Detect incorrect device ID instead of showing the thing as "Online"
Right now any configured thing that has a bridge is shown as "Online", even if the specified device id is invalid (any string possible). We should do a test request to the controller on initialization to check if a device with this ID exists. It might be enough to do this only when the device ID was first set or changed, but as a first step I would do it on each initialization. This can be implemented in the base class `BoschSHCBridgeHandler`

### PR DESCRIPTION
**Pull request by coeing**
Closes #34

### CODE REVIEW

File: bundles/org.openhab.binding.boschshc/src/main/resources/OH-INF/i18n/boschshc.properties [hunk 1]
```diff
@@ -10,3 +10,7 @@ offline.conf-error-ssl = The SSL connection to the Bosch Smart Home Controller i
 offline.long-polling-failed.http-client-null = Long polling failed and could not be restarted because http client is null.
 offline.long-polling-failed.trying-to-reconnect = Long polling failed, will try to reconnect.
 offline.interrupted = Conneting to Bosch Smart Home Controller was interrupted.
+
+offline.conf-error.empty-device-id = No device ID set.
+
+offline.conf-error.invalid-device-id = Device ID is invalid.
```
Comment (GerdZanker) [inline bundles/org.openhab.binding.boschshc/src/main/resources/OH-INF/i18n/boschshc.properties:16]: In the last openhab PR an empty line at EOF was requested.

File: bundles/org.openhab.binding.boschshc/src/main/resources/OH-INF/i18n/boschshc_de.properties [hunk 2]
```diff
@@ -7,3 +7,5 @@ binding.boschshc.description = Dieses Binding integriert das Bosch Smart Home Sy
 offline.conf-error-pairing = Bitte betätigen Sie den Taster am Bosch Smart Home Controller zum automatischen Verbinden.
 offline.not-reachable = Smart Home Controller ist nicht erreichbar.
 offline.conf-error-ssl = Die SSL Verbindung zum Bosch Smart Home Controller ist nicht möglich.
+offline.conf-error.empty-device-id = Keine Geräte ID gesetzt.
+offline.conf-error.invalid-device-id = Geräte ID ist ungültig.
```
Comment (GerdZanker) [inline bundles/org.openhab.binding.boschshc/src/main/resources/OH-INF/i18n/boschshc_de.properties:11]: In the last openhab PR an empty line at EOF was requested.

File: bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/BridgeHandler.java [hunk 3]
```diff
@@ -397,6 +397,45 @@ private boolean getRooms() throws InterruptedException {
...
+        }
+
+        String url = httpClient.getBoschSmartHomeUrl(String.format("devices/%s", deviceId));
+        Request request = httpClient.createRequest(url, GET);
+
+        ContentResponse contentResponse = request.send();
+
+        String content = contentResponse.getContentAsString();
+
+        int statusCode = contentResponse.getStatus();
```
Comment (GerdZanker) [inline bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/BridgeHandler.java:415]: To you see a chance to write one common "sent" error handler to check the status and errorResponse?
We have similar code in ``getRooms`` and ``getDevices`` and for the discovery I again added similar checks.

**Review by GerdZanker — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.boschshc/src/main/resources/OH-INF/i18n/boschshc.properties (hunk 1 above)
Comment (coeing) [reply bundles/org.openhab.binding.boschshc/src/main/resources/OH-INF/i18n/boschshc.properties:16]: It's strange, there is an empty line in my local code and no git changes are detected. Could you check your locally checked out code?

File: bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/BridgeHandler.java (hunk 3 above)
Comment (coeing) [reply bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/BridgeHandler.java]: That's a good idea. I already added the BoschHttpClient.sendRequest method in the past, but then forgot about it. Plus it was a bit too unflexible. I extended the method a bit and use it now in BridgeHandler. You can check if it works in the places you mentioned, maybe we have to extend it even a bit more.

**Review by GerdZanker — APPROVED** [review]
Nice solution to pass the validation logic and the error handler. I will adopt the discovery after merge.

**Pull request merged**